polkadot/runtime/parachains/src/paras/mod.rs

```diff
@@ -802,6 +828,11 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type FutureCodeHash<T: Config> = StorageMap<_, Twox64Concat, ParaId, ValidationCodeHash>;
 
+	/// The code hash authorizations for a para which will expire `expire_at` `BlockNumberFor<T>`.
+	#[pallet::storage]
+	pub type AuthorizedCodeHash<T: Config> =
+		StorageMap<_, Twox64Concat, ParaId, (ValidationCodeHash, BlockNumberFor<T>)>;
```

Comment:
maybe move the data in a struct
```suggestion
		StorageMap<_, Twox64Concat, ParaId, AuthorizedCodeHashAndExpiry<T>>;
```

Reply:
> maybe move the data in a struct

Refactored here: https://github.com/paritytech/polkadot-sdk/pull/7592/commits/cb1dde38f770c18b81088878340005ae88e4a259